src/parser/base.py

```diff
@@ -40,32 +40,25 @@ def handle_execute(self, b_rate_functions, nb_rate_function):
         return result
 
     def safe_parsing(self, func, bank_id):
+        result = None
         try:
             result = func()
             if rate.is_empty(result):
                 raise ParseError("rates are empty")
-
-            return result
-
         except Exception as e:
             self.log.error({
                 "msg": str(e),
                 "country": self.country,
                 "bank_id": bank_id,
             })
-            return None
-
-    def remove_nones(self, rates):
-        ids = []
-
-        for i in rates:
-            if rates[i] is None:
-                ids.append(i)
-                continue
-
-        for i in ids:
-            del rates[i]
+        return result
```

Comment:
Why do you think is better to have variable `result`?

Having `return result` and `return None`  is more explicit and has better readability.